eessi_container.sh

```diff
@@ -489,91 +572,111 @@ fi
+	#   - repo_version,
+	#   - config_bundle, and
+	#   - a map { filepath_in_bundle -> container_filepath }
+        #
+	# The config_bundle includes the files which are mapped ('->') to a target
+	# location in container:
+        # - default.local -> /etc/cvmfs/default.local
+        #   contains CVMFS settings, e.g., CVMFS_HTTP_PROXY, CVMFS_QUOTA_LIMIT, ...
+        # - ${repo_name_domain}.conf -> /etc/cvmfs/domain.d/${repo_name_domain}.conf
+        #   contains CVMFS settings, e.g., CVMFS_SERVER_URL (Stratum 1s),
+        #   CVMFS_KEYS_DIR, CVMFS_USE_GEOAPI, ...
+        # - ${repo_name_domain}/ -> /etc/cvmfs/keys/${repo_name_domain}
+        #   a directory that contains the public key to access the repository, key
+        #   itself then doesn't need to be BIND mounted
+        # - ${repo_name_domain}/${cfg_repo_name}.pub
+        #   (-> /etc/cvmfs/keys/${repo_name_domain}/${cfg_repo_name}.pub
+        #   the public key to access the repository, key itself is BIND mounted
+        #   via directory ${repo_name_domain}
+        cfg_repo_version=$(cfg_get_value ${cfg_repo_id} "repo_version")
+        cfg_config_bundle=$(cfg_get_value ${cfg_repo_id} "config_bundle")
+        cfg_config_map=$(cfg_get_value ${cfg_repo_id} "config_map")
+
+        # convert cfg_config_map into associative array cfg_file_map
+        cfg_init_file_map "${cfg_config_map}"
+        [[ ${VERBOSE} -eq 1 ]] && cfg_print_map
+
+        # use information to set up dir ${EESSI_TMPDIR}/repos_cfg and define
+        #   BIND mounts
+        # check if config_bundle exists, if so, unpack it into
+	#   ${EESSI_TMPDIR}/repos_cfg; if it doesn't, exit with an error
+        # if config_bundle is relative path (no '/' at start) prepend it with
+        #   EESSI_REPOS_CFG_DIR
+        config_bundle_path=
+        if [[ ! "${cfg_config_bundle}" =~ ^/ ]]; then
+            config_bundle_path=${EESSI_REPOS_CFG_DIR}/${cfg_config_bundle}
+        else
+            config_bundle_path=${cfg_config_bundle}
+        fi
 
-  if [[ ! -r ${config_bundle_path} ]]; then
-    fatal_error "config bundle '${config_bundle_path}' is not readable" ${REPOSITORY_ERROR_EXITCODE}
-  fi
+        if [[ ! -r ${config_bundle_path} ]]; then
+            fatal_error "config bundle '${config_bundle_path}' is not readable" ${REPOSITORY_ERROR_EXITCODE}
+        fi
 
-  # only unpack config_bundle if we're not resuming from a previous run
-  if [[ -z ${RESUME} ]]; then
-    tar xf ${config_bundle_path} -C ${EESSI_TMPDIR}/repos_cfg
-  fi
+        # only unpack cfg_config_bundle if we're not resuming from a previous run
+        if [[ -z ${RESUME} ]]; then
+            tar xf ${config_bundle_path} -C ${EESSI_TMPDIR}/repos_cfg
+        fi
 
-  for src in "${!cfg_file_map[@]}"
-  do
-    target=${cfg_file_map[${src}]}
-    BIND_PATHS="${BIND_PATHS},${EESSI_TMPDIR}/repos_cfg/${src}:${target}"
-  done
-  export EESSI_VERSION_OVERRIDE=${repo_version}
-  export EESSI_CVMFS_REPO_OVERRIDE="/cvmfs/${repo_name}"
```

Comment:
Seems like this was important, it shouldn't have been removed.

I'm now seeing errors like this in the build job for https://github.com/EESSI/software-layer/pull/632:
```
ERROR: software.eessi.io/versions/2023.06/compat/linux/aarch64 does not exist!
```

Note the missing `/cvmfs`...

Reply:
Or maybe we're missing a fix in `bot/build.sh`, which now has this:
```shell
export EESSI_CVMFS_REPO_OVERRIDE=$(cfg_get_value "repository" "repo_name")
```

I guess this should be:
```shell
export EESSI_CVMFS_REPO_OVERRIDE=/cvmfs/$(cfg_get_value "repository" "repo_name")
```

Reply:
Ah, yes, good catch, looks like that needs to be changed (and I think it makes more sense that these things are (only) set there and not in the container script).

Reply:
see #633